src/cgutils.cpp

```diff
@@ -2108,6 +2108,9 @@ static Value *boxed(jl_codectx_t &ctx, const jl_cgval_t &vinfo)
         return UndefValue::get(T_prjlvalue);
     if (vinfo.constant)
         return maybe_decay_untracked(literal_pointer_val(ctx, vinfo.constant));
+    // This can happen in early bootstrap for `gc_preserve_begin` return value.
+    if (jt == (jl_value_t*)jl_void_type)
+        return maybe_decay_untracked(literal_pointer_val(ctx, jl_nothing));
```

Comment:
If `vinfo.typ == jl_void_type` then `vinfo.constant == jl_nothing` and we shouldn't get here. What's violating that?

Reply:
https://github.com/JuliaLang/julia/blob/056b374919e11977d5a8d57b446ad1c72f3e6b1d/src/codegen.cpp#L3926 It's not just a `nothing` but also an LLVM `token` though so I'm avoiding touching that.

Reply:
Ah, ok. Right, this was an intentionally malformed value.